source/slang/slang-options.cpp

```diff
@@ -440,59 +465,57 @@ struct OptionsParser
         slang::CompileStdLibFlags compileStdLibFlags = 0;
         bool hasLoadedRepro = false;
 
-        char const* const* argCursor = &argv[0];
-        char const* const* argEnd = &argv[argc];
-        while (argCursor != argEnd)
+        while (reader.hasArg())
         {
-            char const* arg = *argCursor++;
-            if (arg[0] == '-')
-            {
-                UnownedStringSlice argStr = UnownedStringSlice(arg);
+            auto arg = reader.getArgAndAdvance();
+            const auto& argValue = arg.value;
 
-                if(argStr == "-no-mangle" )
+            if (argValue[0] == '-')
+            {
+                if(argValue == "-no-mangle" )
                 {
                     flags |= SLANG_COMPILE_FLAG_NO_MANGLING;
                 }
-                else if (argStr == "-load-stdlib")
+                else if (argValue == "-load-stdlib")
                 {
-                    String fileName;
-                    SLANG_RETURN_ON_FAIL(tryReadCommandLineArgument(sink, arg, &argCursor, argEnd, fileName));
+                    CommandLineArg fileName;
+                    SLANG_RETURN_ON_FAIL(reader.expectArg(fileName));
```

Comment:
It's not a new change here, but we might want to consider what the error-reporting policy for the command line should be. If we liken this to the actual Slang parser (and semantic checking, etc.), there's two principles:

1. Report as many errors as possible in a single pass
2. Do not report "cascading" errors that are actually the byproduct of earlier errors

There's obviously a tension between those two, but right now the use of `SLANG_RETURN_ON_FAIL` during parsing of command-line arguments prioritizes (2) over (1). It almost seems like these cases should `continue` the loop that is processing top-level arguments.